tests/ui/async-await/async-closures/non-copy-arg-does-not-force-inner-move.rs

```diff
@@ -0,0 +1,17 @@
+//@ aux-build:block-on.rs
+//@ edition:2021
+//@ build-pass
+
+#![feature(async_closure)]
+
+extern crate block_on;
+
+fn wrapper(f: impl Fn(String)) -> impl async Fn(String) {
+    async move |s| f(s)
+}
```

Comment:
So, the closure captures `f` by value, and then the future captures the upvar by ref?

Reply:
Yep.